### PR TITLE
chore: mark op1 as deprecated and port most core ops

### PR DESCRIPTION
**Pull request by mmastrac**
Before we remove `#[op]`, give embedders some notice to switch over to `#[op2]`. Note that I could not port the wasm ops yet -- we'll address that in #275

### CODE REVIEW

File: core/examples/hello_world.rs [hunk 1]
```diff
@@ -2,21 +2,12 @@
...
-use deno_core::op;
-use deno_core::Extension;
-use deno_core::JsRuntime;
-use deno_core::Op;
-use deno_core::RuntimeOptions;
-
-// This is a hack to make the `#[op]` macro work with
-// deno_core examples.
-// You can remove this:
 use deno_core::*;
```
Comment (mmastrac) [inline core/examples/hello_world.rs:5]: This is a more convenient import for most of the examples, so leaving it here (it's no longer required though)

File: core/examples/http_bench_json_ops/main.rs [hunk 2]
```diff
@@ -1,26 +1,13 @@
...
-use deno_core::op;
-use deno_core::AsyncRefCell;
-use deno_core::AsyncResult;
-use deno_core::JsBuffer;
-use deno_core::JsRuntimeForSnapshot;
-use deno_core::Op;
-use deno_core::OpState;
-use deno_core::Resource;
-use deno_core::ResourceId;
+use deno_core::*;
```
Comment (mmastrac) [inline core/examples/http_bench_json_ops/main.rs:3]: Ditto

File: core/runtime/tests/ops.rs [hunk 3]
```diff
@@ -288,42 +289,6 @@ fn test_op_detached_buffer() {
     .unwrap();
 }
 
-#[test]
```
Comment (mmastrac) [inline core/runtime/tests/ops.rs:291]: Unused feature, just removing for now

File: core/runtime/tests/snapshot.rs [hunk 4]
```diff
@@ -1,4 +1,5 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+use crate as deno_core;
```
Comment (mmastrac) [inline core/runtime/tests/snapshot.rs:2]: I might remove the `op2(core)` and just do this in the rare files in deno_core that have ops.

Comment (bartlomieju) [reply core/runtime/tests/snapshot.rs:2]: I like this idea

File: core/examples/http_bench_json_ops/main.rs
Comment (bartlomieju) [inline core/examples/http_bench_json_ops/main.rs:1]: We can remove this file altogether I think. It's named `http_bench_json_ops` - it has nothing to do with JSON ops that we did in 2020 or 2021 :D

**Review by bartlomieju — APPROVED** [review]

File: core/runtime/tests/snapshot.rs (hunk 4 above)
Comment (mmastrac) [reply core/runtime/tests/snapshot.rs:2]: I'll follow up with it in a later pass!

**Pull request merged**